amazon/ion/ioncmodule.c

```diff
@@ -1055,14 +1047,14 @@ static iERR ionc_read_timestamp(hREADER hreader, PyObject** timestamp_out) {
                     decContextClearStatus(&dec_context, DEC_Inexact);
                 }
 
-                char* front = "0.";
-                char decimal[fractional_precision];
-                sprintf(decimal, "%d", dec);
-                char* dec_num = string_concatenation(2, fractional_precision, front, decimal);
+                char* dec_num = malloc(fractional_precision + 2);
```

Comment:
Will this be freed when `py_dec_str` below is garbage collected, or does this need to be freed manually after `PyObject* py_dec_str = PyUnicode_FromString(dec_num);` to avoid a memory leak?

Reply:
Ah Thanks.The memory leak test didn't detect any issue and I did some investigation on it. C will eventually de-allocate the memory we allocated after completion of the execution of the program if we don't call free(). However, It's still good to have a free() here in favor of applications that run very long time so we can always free them right after we use them.